### DISCUSSION TITLE
Catching More Than One Type of Exception with One Exception Handler

### DISCUSSION THREAD

**Discussion by mlorek** [dotnet/csharplang#2931] [General]
to support Java like exception handlers, turning the following:

```cs
try{
}
catch (Exception1 ex) {
        throw ex;
}
catch (Exception2 ex) {
        throw ex;
}
```

into:

```cs
try{
}
catch (Exception1 | Exception2 ex) {
        throw ex;
}

```

thanks

**Comment by Joe4evr** [comment]
How do you know which exception was actually caught?

**Comment by HaloFour** [comment]
In Java the type of exception is the most common subtype of the exceptions.  It's not a union in that you can't use any common members between the exception types unless they are accessible on at that most common subtype.  You can achieve the same functionality today in C# with exception filtering:

```cs
try {
    ...
}
catch (Exception ex) when (ex is Exception1 || ex is Exception2) {
    ...
}
```

**Comment by gafter** [comment]
Because you can achieve this today using exception filters in C#, we are unlikely to make a language change.

**Comment by Thaina** [comment]
#344
